core/runtime/src/main/java/io/quarkus/runtime/annotations/RuntimeInitialized.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.runtime.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation that can be used to force a class to be initialized at runtime
+ * when using native image mode. This means that any static init code will be run
+ * at startup, rather than when the image is built.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface RuntimeInitialized {
```

Comment:
It might be a silly question and I'm not totally convinced myself but should we stick to the `Register` prefix?

Reply:
I don't know, I did think about it but I think it reads better this way. 'RegisterAsRuntimeInitialized' could be ok as well.

Reply:
Actually I wonder if all of these should go in a 'native' package, although it would cause problems with the existing RegisterForReflection.